docs/design/release_automation_design.md

```diff
@@ -0,0 +1,72 @@
+## Release Automation Process Design
+
+### Abstract
+
+The release automation process is an automated system that will handle Github
+releases. What this entails is tagging a release, ensure that the release blurb
+contains all the changes that are to go out, and potentially handle release
+artifacts.
+
+### Specification
```

Comment:
Per our previous discussion, I think that regenerating the API client and creating releases of the SDK are pretty orthogonal problems.  I'd be more comfortable moving the whole regeneration issue to a separate design and a separate implementation.

I'm also not convinced that regeneration is enough of a challenge that it warrants automation at this point; spending time building an automation system seems (to me) to be overkill.